### PR TITLE
Oss docker build

### PR DESCRIPTION
**Pull request by max-jacobs**
Reopening this PR for the open source build.

Will squash when merging so we don't have all these testing commits.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -2,16 +2,28 @@ pipeline {
...
     GIT_PASSWORD = "${env.GIT_PSW}"
+
+    DOCKER_REGISTRY = 'expediagroup'
+
+    MVN_OSS_SETTINGS = credentials('eg-oss-settings.xml')
+
+    OSS_GPG_PUB_KEYRING = credentials('pubring.gpg')
+    OSS_GPG_SEC_KEYRING = credentials('secring.gpg')
+    OSS_GPG_KEYNAME='oss@expediagroup.com'
+    OSS_GPG_PASSPHRASE = credentials('private-key-passphrase')
```
Comment (AnanaMJ) [inline Jenkinsfile]: I'll assume there are correct 😆

File: Jenkinsfile [hunk 2]
```diff
@@ -20,15 +32,24 @@ pipeline {
         echo 'Checking out project...'
         checkout scm
         echo 'Building...'
-        withMaven(jdk: 'OpenJDK11', maven: 'Maven3.6', mavenSettingsConfig: 'hcomdata-artifactory-maven-settings') {
-          sh 'mvn clean deploy jacoco:report checkstyle:checkstyle'
+        sh 'mvn clean deploy jacoco:report checkstyle:checkstyle -s ${MVN_OSS_SETTINGS}'
```
Comment (massdosage) [inline Jenkinsfile]: We should add spotbugs in here (can be a separate PR)

Comment (max-jacobs) [reply Jenkinsfile]: I did try this, but got a maven error saying it couldn't be found. Will make this a separate PR.

Comment (massdosage) [reply Jenkinsfile]: Yes, you need to use the latest oss-parent (release this morning!) to get it.
